bundledatasrc_test.go

```diff
@@ -46,77 +67,40 @@ applications:
         acl:
           admin: "admin"
           foo: "consume"
-`
-)
-
-func (s *BundleDataSourceSuite) TestReadMultiDocBundle(c *gc.C) {
-	specs := []struct {
-		descr     string
-		getSource func() (BundleDataSource, error)
-	}{
-		{
-			descr: "reading bundle from local yaml file",
-			getSource: func() (BundleDataSource, error) {
-				path := bundleDirPath(c, "wordpress-multidoc")
-				return LocalBundleDataSource(filepath.Join(path, "bundle.yaml"))
-			},
-		},
-		{
-			descr: "reading bundle from exploded archive folder",
-			getSource: func() (BundleDataSource, error) {
-				path := bundleDirPath(c, "wordpress-multidoc")
-				return LocalBundleDataSource(path)
-			},
-		},
-		{
-			descr: "reading bundle from archive",
-			getSource: func() (BundleDataSource, error) {
-				path := archiveBundleDirPath(c, "wordpress-multidoc")
-				return LocalBundleDataSource(path)
-			},
-		},
-		{
-			descr: "reading bundle from stream",
-			getSource: func() (BundleDataSource, error) {
-				r := strings.NewReader(multidocBundle)
-				return StreamBundleDataSource(r, "https://example.com")
-			},
-		},
-	}
-
-	for specIndex, spec := range specs {
-		c.Logf("[test %d] %s", specIndex, spec.descr)
+`)
 
-		ds, err := spec.getSource()
-		c.Assert(err, gc.IsNil)
+	src, err := StreamBundleDataSource(r, "https://example.com")
+	c.Assert(err, gc.IsNil)
+	assertBundleSourceProcessed(c, src)
+}
 
-		parts := ds.Parts()
-		c.Assert(parts, gc.HasLen, 3)
-		s.assertFieldPresent(c, parts[1], "applications.wordpress.offers.offer1.endpoints")
-		s.assertFieldPresent(c, parts[2], "applications.wordpress.offers.offer1.acl.admin")
-	}
+func assertBundleSourceProcessed(c *gc.C, src BundleDataSource) {
+	parts := src.Parts()
+	c.Assert(parts, gc.HasLen, 3)
+	assertFieldPresent(c, parts[1], "applications.wordpress.offers.offer1.endpoints")
+	assertFieldPresent(c, parts[2], "applications.wordpress.offers.offer1.acl.admin")
 }
 
 func (s *BundleDataSourceSuite) TestResolveAbsoluteFileInclude(c *gc.C) {
 	target, err := filepath.Abs(filepath.Join(c.MkDir(), "example"))
 	c.Assert(err, gc.IsNil)
 
-	expContent := []byte("example content")
+	expContent := []byte("example content\n")
 	c.Assert(ioutil.WriteFile(target, expContent, os.ModePerm), gc.IsNil)
 
 	ds := new(resolvedBundleDataSource)
 
 	got, err := ds.ResolveInclude(target)
 	c.Assert(err, gc.IsNil)
-	c.Assert(got, gc.DeepEquals, expContent)
+	c.Assert(string(got), gc.DeepEquals, string(expContent))
 }
 
 func (s *BundleDataSourceSuite) TestResolveRelativeFileInclude(c *gc.C) {
 	relTo := c.MkDir()
 	target, err := filepath.Abs(filepath.Join(relTo, "example"))
 	c.Assert(err, gc.IsNil)
 
-	expContent := []byte("example content")
+	expContent := []byte("example content\n")
```

Comment:
doesn't need to be changed, but I would usually have the 'canonical' form as a string, and then convert to []byte only where you need it. eg:

expContent := "example content\n"
cAssert(ioutil.WriteFile(target, []byte(expContent), os.ModePerm), gc.IsNil)

...
c.Assert(string(got), gc.Equals, expContent)

(don't need DeepEquals for strings)

Reply:
Wow, how did I miss that? Fixed now!